doc/source/ray-air/computer-vision.rst

```diff
@@ -0,0 +1,285 @@
+
+    ham
+
+Creating checkpoints
+--------------------
+
+:class:`Checkpoints <ray.air.checkpoint.Checkpoint>` are required for batch inference and model
+serving. They contain model state and optionally a preprocessor.
+
+.. tabbed:: Torch
+
+    To create a :class:`~ray.train.torch.TorchCheckpoint`, pass a Torch model and
+    the :class:`~ray.data.preprocessor.Preprocessor` you created in `Transforming images`_
+    to :meth:`TorchCheckpoint.from_model() <ray.train.torch.TorchCheckpoint.from_model>`.
+
+    .. literalinclude:: ./doc_code/torch_computer_vision.py
+        :start-after: __torch_checkpoint_start__
+        :end-before: __torch_checkpoint_stop__
+        :dedent:
+
+.. tabbed:: TensorFlow
+
+    ham
+
+.. tip::
+    :meth:`Trainer.fit() <ray.train.trainer.BaseTrainer.fit>` returns a :class:`~ray.air.result.Result` object.
+    If you're going from training to prediction, don't create a new checkpoint. Instead,
+    use :attr:`Result.checkpoint <ray.air.result.Result.checkpoint>`.
+
+Batch predicting images
+-----------------------
+
+:class:`~ray.train.batch_predictor.BatchPredictor` lets you perform inference on large
+image datasets.
+
+.. tabbed:: Torch
+
+    To create a ``BatchPredictor``, call
+    :meth:`BatchPredictor.from_checkpoint <ray.train.batch_predictor.BatchPredictor.from_checkpoint>` and pass the checkpoint
+    you created in `Creating checkpoints`_.
+
+    .. literalinclude:: ./doc_code/torch_computer_vision.py
+        :start-after: __torch_batch_predictor_start__
+        :end-before: __torch_batch_predictor_stop__
+        :dedent:
+
+    For a more in-depth example, read :doc:`/ray-air/examples/pytorch_resnet_batch_prediction`.
+
+.. tabbed:: TensorFlow
+
+    ham
+
+Serving vision models
+---------------------
+
+:class:`~ray.serve.air_integrations.PredictorDeployment` lets you
+deploy a model to an endpoint and make predictions over the Internet.
+
+Deployments use :ref:`HTTP adapters <serve-http>` to define how HTTP messages are converted to model
+inputs. For example, :func:`~ray.serve.http_adapters.json_to_ndarray` converts HTTP messages like:
+
+.. code-block::
+
+    {"array": [[1, 2], [3, 4]]}
+
+To a NumPy ndarrays like:
+
+.. code-block::
+
+    array([[1., 2.],
+            [3., 4.]])
+
+
+.. tabbed:: Torch
+
+    To deploy a Torch model to an endpoint, pass the checkpoint you created in `Creating checkpoints`_
+    to :meth:`PredictorDeployment.bind <ray.serve.air_integrations.PredictorDeployment.bind>` and specify
+    :func:`~ray.serve.http_adapters.json_to_ndarray` as the HTTP adapter.
+
+    .. literalinclude:: ./doc_code/torch_computer_vision.py
+        :start-after: __torch_serve_start__
+        :end-before: __torch_serve_stop__
+        :dedent:
+
+    Then, make a request to classify an image.
+
+    .. literalinclude:: ./doc_code/torch_computer_vision.py
+        :start-after: __torch_online_predict_start__
+        :end-before: __torch_online_predict_stop__
+        :dedent:
+
```

Comment:
let's also link to "Deploying Predictors with Serve" user guide and a full example (like the mnist image classifier one)